avogadro/core/mesh.cpp

```diff
@@ -177,5 +178,95 @@ Mesh& Mesh::operator=(const Mesh& other)
   return *this;
 }
 
+void Mesh::smooth()
+{
+  if (m_vertices.size() == 0)
+    return;
+
+  // Map vertices to a line and pass them to NeighborPerceiver
+  Array<Vector3> linearList(m_vertices.size());
+  for (size_t i = 0; i < m_vertices.size(); i++)
+    linearList[i] = Vector3(
+      double(m_vertices[i](0) + 1.31*m_vertices[i](1) + 0.97*m_vertices[i](2)),
```

Comment:
I understand the concept, but not the particular constants?

Reply:
No reason in particular... I just tried random values near 1 with a `std::cout << neighbors.size() << "\n"` statement to make the distribution as homogeneous as possible. I should probably comment an explanation...